File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    cloth_dict = {}
    for i in clothes:
        if i[1] not in cloth_dict:
            cloth_dict[i[1]] = 1
        elif i[1] in cloth_dict:
            cloth_dict[i[1]] += 1
    for i in cloth_dict:
        answer *= (cloth_dict[i]+1)
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        for(int i = 0; i < clothes.length; i++){
            String type = clothes[i][1];
            map.put(type, map.getOrDefault(type, 0)+1);
        }
        
        for(String type: map.keySet()){
            // System.out.println(type+" : "+map.get(type));
            answer *= map.get(type)+1;
        }
        return answer-1;
    }
}